### ISSUE TITLE
Error when building in parallel: `sh: 1: cargo: not found`

### ISSUE THREAD

**Issue by probablykasper** [cross-rs/cross#496]
When running `cross` in multiple instances at once, the following error shows up:
```
sh: 1: cargo: not found
```

If this is too difficult to fix, there could be an improved error, or a warning.

**Comment by TarekkMA** [comment]
I'm getting this error on GitHub action, I didn't run it in parallel

**Comment by cessen** [comment]
I'm getting this error just with a plain `cross build` as well.  Running NixOS.  Seemingly can't use `cross` because of this error.

**Comment by cessen** [comment]
It looks like this might be related to (or the same as) #260

**Comment by Alexhuszagh** [comment]
If you're running NixOS, then yes it should be a duplicate of #260, but I don't believe the original error is. I'll see if I can replicate this in parallel.

**Comment by Alexhuszagh** [comment]
This seems to be a more general Docker issue. Try the following and Docker will deadlock:

```bash
for i in {1..5}; do
    docker run -it --rm ubuntu:20.04 bash -c "ls" &
done
```

**Comment by Alexhuszagh** [comment]
I can't replicate this anymore, and `cross` doesn't use global state and only calls `cargo` and `docker` which have good consistency mechanisms.
